Source/WebKit/Shared/Cocoa/CoreIPCPlistDictionary.mm

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "CoreIPCPlistDictionary.h"
+
+#if PLATFORM(COCOA)
+
+#import "CoreIPCData.h"
+#import "CoreIPCDate.h"
+#import "CoreIPCNumber.h"
+#import "CoreIPCPlistArray.h"
+#import "CoreIPCPlistObject.h"
+#import "CoreIPCString.h"
+
+namespace WebKit {
+
+CoreIPCPlistDictionary::CoreIPCPlistDictionary(NSDictionary *dictionary)
+{
+    m_keyValuePairs.reserveInitialCapacity(dictionary.count);
+
+    for (id key in dictionary) {
+        id value = dictionary[key];
+
+        if (!key || ![key isKindOfClass:[NSString class]])
+            continue;
+
+        if (!CoreIPCPlistObject::isPlistType(value))
+            continue;
```

Comment:
Maybe put an ASSERT_NOT_REACHED on these so we might catch any non-plist contents we would be discarding in release builds.